interfaces/adfsuite/ams.py

```diff
@@ -1041,8 +1062,33 @@ def get_force_constants(self, engine: Optional[str] = None) -> np.ndarray:
         forceConstants = np.array(forceConstants) if isinstance(forceConstants, list) else np.array([forceConstants])
         return forceConstants
 
-    def get_normal_modes(self, engine: Optional[str] = None) -> np.ndarray:
+    def get_pvdos(self, engine: Optional[str] = None):
+        """Return a numpy array of Partial Vibrational Spectra (PVDOS) with shape: (nNormalModes, nAtoms), with values [0,1].
+
+        The *engine* argument should be the identifier of the file you wish to read. To access a file called ``something.rkf`` you need to call this function with ``engine='something'``. The *engine* argument can be omitted if there's only one engine results file in the job folder.
+        """
+        pvdos = self._process_engine_results(lambda x: x.read("Vibrations", "PVDOS"), engine)
+        nNormalModes = self._process_engine_results(lambda x: x.read("Vibrations", "nNormalModes"), engine)
+        nAtoms = len(self.job.molecule)
+        pvdos = np.array(pvdos).reshape(nNormalModes, nAtoms)
+        return pvdos
+
+    def get_reduced_masses(self, engine: Optional[str] = None):
+        """Return a numpy array of reduced masses, expressed in amu units.
+        If mass_weighted_hessian_eigenvectors=True it returns the mass_weighted_hessian_eigenvectors.
+
+        The *engine* argument should be the identifier of the file you wish to read. To access a file called ``something.rkf`` you need to call this function with ``engine='something'``. The *engine* argument can be omitted if there's only one engine results file in the job folder.
+        """
+        reduced_masses = np.array(
+            self._process_engine_results(lambda x: x.read("Vibrations", "ReducedMasses"), engine)
+        )
+        return reduced_masses
+
+    def get_normal_modes(
+        self, mass_weighted_hessian_eigenvectors: Optional[bool] = False, engine: Optional[str] = None
```

Comment:
Additional keyword arguments should go at the end to avoid breaking change to method signature e.g.
```
get_normal_modes("adf")
```
would no longer work